lib/charms/prometheus_k8s/v1/prometheus.py

```diff
@@ -1,36 +1,312 @@
+"""
+## Overview
+
+This document explains how to integrate with the Prometheus charm
+for the purposes of providing a metrics endpoint to Prometheus. It
+also explains how alternative implementations of the Prometheus charm
+may maintain the same interface and be backward compatible with all
+currently integrated charms. Finally this document is the
+authoritative reference on the structure of relation data that is
+shared between Prometheus charms and any other charm that intends to
+provide a scrape target for Prometheus.
+
+## Consumer Library Usage
+
+This Prometheus charm interacts with its scrape targets using this
+charm library This charm library is constructed using the [Provider
+and Consumer](https://ops.readthedocs.io/en/latest/#module-ops.relation)
+objects from the Operator Framework. This implies charms that would
+like to expose metric endpoints for the Prometheus charm must use the
+`PrometheusConsumer` object from the charm library to do so. Using the
+`PrometheusConsumer` object requires instantiating it, typically in
+the constructor of your charm (the one which exposes the metrics
+endpoint). The `PrometheusConsumer` constructor requires the name of
+the relation over which a scrape target (metrics endpoint) is exposed
+to the Promtheus charm. This relation must use the `prometheus_scrape`
+interface. In addition the constructor also requires a `consumes`
+specification, which is a dictionary with key `prometheus` (also see
+Provider Library Usage below) and a value that represents the minimum
+acceptable version of Prometheus. This version string can be in any
+format that is compatible with Python [Semantic Version
+module](https://pypi.org/project/semantic-version/).  For example,
+assuming your charm exposes a metrics endpoint over a relation named
+"monitoring", you may instantiate `PrometheusConsumer` as follows
+
+    from charms.prometheus_k8s.v0.prometheus import PrometheusConsumer
```

Comment:
You've mostly been using triple backticks in other documentation - may be worth being consistent here.

Reply:
I have not typically come across backticks in Python doc strings and hence find that unusual. This may be because backticks only look nice when they are replaced by shaded backgrounds on being rendered as markdown. However backticks within the Python help system `help()` would only add clutter with no benefit.

Reply:
Ah yes, you could be right.